[doc] update distributed optimizer doc

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#58084 [doc] update distributed optimizer doc**

update the doc for distributed optimizer with TorchScript support.

Differential Revision: [D28363971](https://our.internmc.facebook.com/intern/diff/D28363971)

Commit:
[doc] update distributed optimizer doc

update the doc for distributed optimizer with TorchScript support.

[ghstack-poisoned]

rohan-varma commented on torch/distributed/optim/optimizer.py:
```diff
@@ -154,10 +154,10 @@ class DistributedOptimizer:
-    are increasing the coverage to all optimizers in future releases.
+    Interpreter Lock (GIL) in the case of multithreaded training (e.g. Distributed
+    Model Parallel). This feature is currently enabled for most optimizers. You
+    can also follow the recipe in PyTorch tutorials to enable TorchScript support
```
Nit: link the recipe? :)

wanchaol replied:
I tried but I don't know how I can link it though.. since the recipe don't have a dev page and it will be published together with 1.9, so we don't have a link now.. Do you know how we could get a link before the release? Or should we link a deploy preview as [this](https://deploy-preview-1465--pytorch-tutorials-preview.netlify.app/) or simply the PR link?

Commit:
Update on "[doc] update distributed optimizer doc"


update the doc for distributed optimizer with TorchScript support.

Differential Revision: [D28363971](https://our.internmc.facebook.com/intern/diff/D28363971)

[ghstack-poisoned]

wanchaol commented:
@wanchaol has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28363971).
